validator.go

```diff
@@ -458,13 +458,13 @@ func IsIP(str string) bool {
 // IsIPv4 check if the string is an IP version 4.
 func IsIPv4(str string) bool {
 	ip := net.ParseIP(str)
-	return ip != nil && ip.To4() != nil
+	return ip != nil && strings.Contains(str, ".")
```

Comment:
`::FFFF:127.0.0.1` is a valid IPv6 address that contains `.`

You could swap it for `return ip != nil && !strings.Contains(str, ":")`.

Reply:
If ports could be appended at the end, both method would need to be changed to `strings.Count(str, ":") < 2` and `strings.Count(str, ":") >= 2` respectively.